hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -126,54 +131,51 @@ private void removeDeadDataNodes() {
   }
 
   public List<String> getDeadJournalNodes() {
+    List<String> deadJournalHosts = new ArrayList<>();
+
     if (deadJournalNodeTimeStamp != null && deadJournalNodeTimeStamp.before(new Date())) {
       removeDeadJournalNodes();
-      return new ArrayList<>();
     } else {
       Map<String, String> journalNodes = getJournalNodes();
-      Set<String> journalHosts = journalNodes.keySet();
-      List<String> deadJournalHosts = new ArrayList<>();
-      for (String journalHost : journalHosts) {
-        if (journalNodes.get(journalHost) == null) {
-          deadJournalHosts.add(journalHost);
```

Comment:
I would prefer to see a separate variable here to avoid method chaining or too many method calls on the same line. This is a nit and a style choice, but generally we want to avoid too many errors occurring on the same line. We try to do that throughout the project to avoid complexity.
 
This is what I am looking for:

Set<Map.Entry<String, String>> journalNodeEntries = journalNodes.entrySet();
for (Map.Entry<String, String> journalNode : journalNodeEntries)